apps/api-journeys/src/app/modules/journey/journey.resolver.ts

```diff
@@ -160,6 +182,65 @@ export class JourneyResolver {
     })
   }
 
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyArchive(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.archived,
+      archivedAt: new Date().toISOString()
+    })
+  }
+
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyDelete(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.deleted,
+      deletedAt: new Date().toISOString()
+    })
+  }
+
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyTrash(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.trashed,
+      trashedAt: new Date().toISOString()
+    })
+  }
+
+  async journeyTrashAllArchived(
+    @CurrentUserId() userId: string
+  ): Promise<Journey[]> {
+    const results: Journey[] = (await this.journeyService.getAllByOwner(userId))
+      .filter((j) => j.status === JourneyStatus.archived)
+      .map((j) => ({
+        ...j,
```

Comment:
small nit. Not a huge fan of shorthand.
```suggestion
      .filter((journey) => journey.status === JourneyStatus.archived)
      .map((journey) => ({
        ...journey,
```